components/hacbs/jvm-build-service/overlays/dev/apiexport.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: apis.kcp.dev/v1alpha1
+kind: APIExport
+metadata:
+  name: jvm-build-service
+spec:
+  permissionClaims:
+    - group: ""
+      resource: "secrets"
+    - group: ""
+      resource: "serviceaccounts"
+    - group: ""
+      resource: "events"
+    - group: ""
+      resource: "services"
+      identityHash: pipeline-service
+    - group: "apps"
+      resource: "deployments"
+      identityHash: pipeline-service
```

Comment:
why do you claim `services` and `deployments`? they shouldn't be available nor used in the main appstudio user workspace.

Reply:
JVM build service is creating cache deployment/service per each namespace. As part of kcp transition the cache could be potentially created on workspace level (or maybe configurable).

Reply:
Well, the plan is that the "main" AppStudio workspace won't have this API available at all. Deployments will be available only in environment sub-workspaces. 

As of now, the Deployment API is imported as part of the pipeline-service APIBinding - I discussed this in the last Arch call with Romain and they are aware of that and they want to address it in this sprint. They want to split their APIExports into two of them, where one will contain only the Tekton API which will be bound in the user AppStudio workspace. This means that the Deployment nor Service API won't be available in AppStudio user workspace at all.
If you still need that, then it has to be configured as part of the HACBS stack. Not sure which workload cluster would you use for that...

Reply:
Yes, we are aware of API Export split (requested by us). @gabemontero @stuartwdouglas do we have a plan which workload cluster will be used for deployment of caches?

Reply:
Yep I started asking the question of how should the artifact cache of the jvm-build-service be deployed last week (both in the placement service cabal and hacbs weekly).  We have received some responses to that question but I would say finalizing the answer is not complete yet.  There are also some next level details around confirming which workload/compute clusters we should expect various API objects to be sync'ed to, where again we've received some responses but need another round or so of clarification.  i.e. with routing updates (see below) in place, we could function with the artifact cache in one workload/compute cluster, and the pipeline runs, jvm build service custom resources instances, and our associated k8s objects like configmaps and secrets co-located in another workload/compute cluster.  But any more flexibility will become very very very hard for us to overcome. 

So, current status / "plan" as I understand it:
- the current form of the jvm-build-service controller creating the deployment for the artifact cache as part of HACBS workspace/namespace set up, i.e. after a user / workspace admin has created an instance of our `UserConfig` CR, is appearing "not possible".  But we won't remove that support from the controller until we are really really sure.
- the on going infra deployment work will get our controller deployed, and ultimately address our permission claims so that our controller can watch/access `PipelineRuns` and `TaskResults`.  And yes, there is some appstudio vs. HACBS distinctions there.
- however, completing the configuration of the "HACBS stack", including defining / deploying the jvm-build-servivce artifact cache Deployment and setting up the necessary ingress and haproxy so that customer builds can access that cache, is beyond the role of infra deployments, and will have to be done as part of HACBS specific activity in the workspace settings controller the infra team recently exposed.

But we need to continue to have conversations with various SMEs on the other teams to iron all this out.

And I already alluded to it above, but to be clear, aside from just getting the artifact cache deployed, we also have the question of how a customer's build running via a `PipelineRun` accesses the artifact cache.  I *think* the answer to that one is a bit clearer (i.e. ingress